docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,595 @@
+from the current route advertisement behavior, the routes to remote node subnets
+will be imported to the Gateway Router. Once the traffic reach the
+GatewayRouter, it will be forwarded the nexthop node according to routes learnt
+from BGP.
+
+In Local Gateway Mode, the pod to remote pod traffic egress traffic will enter
+the host VRF via the ovn-k8s-mpX interface. Then it will be forwarded to remote
+nodes according to the host routes of the VRF.
+
+The following diagram shows the OVN topology of a cluster with three networks:
+
+* a CUDN (in green) that uses Geneve transport (default behavior)
+* a default network (in gray) that enables no-overlay mode
+* a CUDN (in blue) that enables no-overlay mode
+
+![No-overlay topology](../images/no_overlay_topology.jpg)
+
+#### No SNAT in no-overlay mode for egress traffic
+
+The BGP feature in ovn-kubernetes only disables SNAT for traffic that is
+destined to external networks, not for traffic that is destined to other pods in
+the cluster.
+
+In no-overlay mode, we want to fully leverage BGP and avoid the SNAT step on the
+gateway router for intra-cluster traffic as well. Pod-to-pod traffic within the
+same network will hit the BGP-learned routes to reach its destination node,
+without being encapsulated in a Geneve tunnel and without being SNATed to the
+node IP.
+
+The only exception is for pod to other host traffic, to ensure nodePort services
+can be accessed cross networks. The egress traffic will be SNATed to the node IP
+before leaving the node.
+
+#### Isolation between UDNs with no-overlay mode enabled
+
+The [existing UDN isolation
+mechanism](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5186/commits/8f6e7d30ee5f4926a21e2de75488aad80344814b)
+will still be in place. Pods in different UDNs will be isolated from each other
+and from the default network by means of the existing OVN ACLs on every node
+logical switch: ACLs are enforced on outgoing traffic at each advertised UDN
+switch, verifying whether both source and destination IPs belong to the same
+advertised UDN. If the destination IP does not come from the same UDN subnet as
+the source IP, traffic is dropped. The no-overlay mode will not affect the
+isolation between UDNs or between UDNs and the default network.
+
+### Workflow
+
+#### Enable No-overlay Mode for the Default Network
+
+No-overlay mode for the default network must be enabled before OVN-Kubernetes
+pods start. Therefore, the following configuration manifests shall be created
+before deploying the OVN-Kubernetes pods.
+
+1. The cluster admin enables no-overlay mode by creating the well-known
+   `default` UDN CR with `spec.layer3.encapsulation` set to `None`.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: ClusterUserDefinedNetwork
+    metadata:
+      # 'default' is the reserved name to represent the cluster default network
+      name: default
+    spec:
+      # use an empty selector to select all namespaces
+      namespaceSelector: {}
```

Comment:
Should we invalidate all other values for `namespaceSelector`?

Reply:
Maybe we can add a validation rule for this.